// actionHandlers/searchOccupations.js
var api = require('api');
var getContext = require('helpers/getContext');
var getTilybotPayload = require('helpers/getTilybotPayload');

const PAGE_SIZE = 10;

var action = 'searchOccupations';
async function handler(req, res, next) {
	let payload = getTilybotPayload(req.body.result.fulfillment.messages);
	let credsContext = getContext("creds", req.body.result.contexts);

	let tilrRes = await api.get('/occupations/search/' + req.body.result.resolvedQuery);
	if (tilrRes.data.occupations.length > 0) {
		req.log.info('found results for query "%s"', req.body.result.resolvedQuery);
		let occupations = tilrRes.data.occupations.slice(0, PAGE_SIZE <= tilrRes.data.occupations.length ? PAGE_SIZE : tilrRes.data.occupations.length);
		res.send({
			data: {
				tilybot: {
					messages: [
						...payload.messages,
						{
							type: 'buttons-stack',
							options: occupations.map((elem) => {
								return { value: elem.id, label: elem.title };
							})
						}
					]
				}
			},
			contextOut: [
				{
					name: 'occupations-success',
					lifespan: 2
				}
			]
		});
	} else {
		req.log.info('couldn\'t find results for query "%s"', req.body.result.resolvedQuery);
		res.send({ followupEvent: { name: 'OCCUPATIONS_NO_RESULTS' }, contextOut: [{ name: 'occupations-fail', lifespan: 1 }]});
	}
}

module.exports = { action, handler };
