// actionHandlers/setJobFamily.js
var api = require('api');
var getContext = require('helpers/getContext');
var getTilybotPayload = require('helpers/getTilybotPayload');

const ID = "de174312-199c-4629-93a7-a63eeda97f03";

var action = 'setJobFamily';
async function handler(req, res, next) {
	let payload = getTilybotPayload(req.body.result.fulfillment.messages);
	let responseContext = getContext("jobfamily-response-followup", req.body.result.contexts);
	let credsContext = getContext("creds", req.body.result.contexts);

	await api.post(
		'/vetting_questions/' + ID + '/answer',
		{ answer: { answer: responseContext.parameters.categories.join(", ") } },
		{
			headers: {
				'X-API-EMAIL': credsContext.parameters.email,
				'X-API-TOKEN': credsContext.parameters.token
			}
		}
	);
	req.log.info('posted response "%s"', responseContext.parameters.categories.join(", "));
	res.send({
		contextOut: [
			{
				name: 'jobfamily-response-followup',
				lifespan: 0
			}
		],
		data: { tilybot: payload }
	});
}

module.exports = { action, handler };
