// actionHandlers/setTravelMethod.js
var api = require('api');
var getContext = require('helpers/getContext');
var getTilybotPayload = require('helpers/getTilybotPayload');

const ID = "fe59af48-55cb-49f3-b8c4-9f4c01af641b";

var action = 'setTravelMethod';
async function handler(req, res, next) {
	let payload = getTilybotPayload(req.body.result.fulfillment.messages);
	let responseContext = getContext("transportation-response-followup", req.body.result.contexts);
	let credsContext = getContext("creds", req.body.result.contexts);

	await api.post(
		'/vetting_questions/' + ID + '/answer',
		{ answer: { answer: responseContext.parameters.travelMethods.join(", ") } },
		{
			headers: {
				'X-API-EMAIL': credsContext.parameters.email,
				'X-API-TOKEN': credsContext.parameters.token
			}
		}
	);
	req.log.info('posted response "%s"', responseContext.parameters.travelMethods.join(", "));
	res.send({
		contextOut: [
			{
				name: 'transportation-response-followup',
				lifespan: 0
			}
		],
		data: { tilybot: payload }
	});
}

module.exports = { action, handler };
